monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -388,13 +388,21 @@ def get_config_users():
     def get_config_passwords():
         return ConfigService.get_config_value(PASSWORD_LIST_PATH, True)
 
-    @staticmethod
-    def get_config_exploits():
-        exploits_config_value = EXPLOITER_CLASSES_PATH
-        exploits = ConfigService.get_config_value(exploits_config_value, True)
+    @classmethod
+    def get_config_exploits(cls):
+        agent_configuration = cls._agent_configuration_repository.get_configuration()
+        exploitation_configuration = agent_configuration.propagation.exploitation
+
+        enabled_exploiters = (
+            exploiter
+            for exploiter in chain(
+                exploitation_configuration.brute_force, exploitation_configuration.vulnerability
+            )
+        )
```

Comment:
Why is this better than simply doing `enabled_exploiters = list(chain(exploitation_configuration.brute_force, exploitation_configuration.vulnerability))`?

Reply:
(Also relevant for `get_config_users()` and `get_config_passwords()`)

Reply:
In this case, doing `list()` is unnecessary. Creating the iterator is more efficient (not that efficiency really matters in this case). But you're right, all we need is `chain()`.